### ISSUE TITLE
Switch between markdown and text editor

### ISSUE THREAD

**Issue by jooola** [nextcloud/text#323]
**Is your feature request related to a problem? Please describe.**
When editing markdown files, I sometimes need to edit the raw (not parsed) markdown content, instead of the rendered one. This option is not available currently.

**Describe the solution you'd like**
A button near the toolbar could allow someone to switch from the markdown rich editor to the basic text editor in order to edit the markdown file in raw mode.

**Comment by juliushaertl** [comment]
Duplicate of #123 see especially  https://github.com/nextcloud/text/issues/123#issuecomment-508723049 for some comments on the feature request.

**Comment by jooola** [comment]
I don't think this is a duplicate, I don't want to see both the markdown source and the rendered html, I would like to swap **between** the the rich text and plain text editor. At least this is the differences I see between #123 and this request.

You mentioned that this could be difficult to handle with the collaborative editing.  Not sure how it works in this app, but I made few research on how we could enable this, and it seems already supported by ProseMirror.

I found this : https://prosemirror.net/examples/markdown/

And it should be easy to integrate (again I don't know how the collaborative editing works with this, simply suggesting)

Cheers

**Comment by juliushaertl** [comment]
> You mentioned that this could be difficult to handle with the collaborative editing. Not sure how it works in this app, but I made few research on how we could enable this, and it seems already supported by ProseMirror.

The prose mirror example always creates a new editor when switching, which does not work with collaborative editing. Note that even if we use the same markdown file, the document representation format that is used by prosemirror will be different for visual and plain text editing.

The main issue here is that when switching all collaborative history data is lost, so if once connected client is not fully up to date or has unsaved changes, they would be lost once we switch the editing mode. Also switching cannot be done per user but would be per document due to the document representation format difference stated above.

I hope that helps to understand why I think adding such a feature will cause more issues than it solves.

**Comment by jooola** [comment]
Ok, got it ! Thanks. Sorry for the trouble.

**Comment by juliushaertl** [comment]
No problem, I get that there is a use case for it, it's just nothing that is likely to become implemented due to technical limitations and the primary goal we had when creating the text app.

**Comment by small1** [comment]
There is a problem at the moment that all markdown files opens by default in the text app. If you want to edit the source you have to rename it to a txt file then edit and then rename it back. 

One solution is to have the ability to switch modes but do it with a warning "this will destroy collaboration and history" 

The other has to be done at server level i guess. To have the ability to open the markdown file as plain text or as markdown. That could work as well. Just not having to rename the file for a simple edit (editing files for cms_pico for example as inerting an image point to an asset url that text cant read)

**Comment by laarisyko** [comment]
@small1 I'm a supporter of you point. Pico CMS is great tool, because of its seemless integration and ease of use. The current text editor is great for simple stuff, but it would be great to have option without renaming the files to change the raw file. Even open by the details dialog would be fine. We have for example several files with endings like .dat or similar, which are in the end just text files. I would be great to have the option to open these files with the text editor. 

What acutally happend to the old markdown editor, could it be used in parallel to the new text editor?

**Comment by claell** [comment]
Duplicate of #123
